File: packages/api/src/lib/task-handlers.ts
import { registerHandler } from "task-queue";
import { verifyFileHandler } from "./verify.worker";
import { checkFileHandler } from "./check.worker";
import { loadAssetsHandler } from "./load-assets.worker";
import { uploadAssetHandler } from "./upload-asset.worker";
import { prefetchDatabaseHandler } from "./prefetch-database.worker";
import { syncDatabaseHandler } from "./sync-database.worker";
import { saveAssetHandler } from "./save-asset.worker";
import { saveAssetsBatchHandler } from "./save-assets-batch.worker";
import { createDatabaseHandler } from "./create-database.worker";
import { importAssetsHandler } from "./import-assets.worker";
import { hashFileHandler } from "./hash-file.worker";

//
// Register all task handlers
// This has to be called from the worker thread.
//
export function initTaskHandlers(): void {
    registerHandler("verify-file", verifyFileHandler);
    registerHandler("check-file", checkFileHandler);
    registerHandler("load-assets", loadAssetsHandler);
    registerHandler("prefetch-database", prefetchDatabaseHandler);
    registerHandler("upload-asset", uploadAssetHandler);
    registerHandler("sync-database", syncDatabaseHandler);
    registerHandler("save-asset", saveAssetHandler);
    registerHandler("save-assets-batch", saveAssetsBatchHandler);
    registerHandler("create-database", createDatabaseHandler);
    registerHandler("import-assets", importAssetsHandler);
    registerHandler("hash-file", hashFileHandler);
}

File: packages/node-api-zig/src/lib/task-handlers.zig
const task_queue_zig = @import("task-queue-zig");
const verify_worker = @import("verify.worker.zig");
const replicate_database_worker = @import("replicate-database.worker.zig");
const registerHandler = task_queue_zig.worker.registerHandler;
const verifyFileHandler = verify_worker.verifyFileHandler;
const replicateDatabaseHandler = replicate_database_worker.replicateDatabaseHandler;

//
// Register all task handlers
// This has to be called from the worker thread.
// (Zig: worker threads share the process-wide handler registry, so this is called once at startup.)
//
pub fn initTaskHandlers() !void {
    try registerHandler("verify-file", verifyFileHandler);
    // Not ported: check-file, load-assets, prefetch-database, upload-asset, sync-database (not used by psi replicate or psi verify).
    try registerHandler("replicate-database", replicateDatabaseHandler);
    // Not ported: save-asset, save-assets-batch, create-database, import-assets, hash-file, get-database-summary,
    // move-assets (not used by psi replicate or psi verify).
}
